pkg/dogstatsd/parse_events.go

```diff
@@ -0,0 +1,192 @@
+package dogstatsd
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type eventPriority int
+
+const (
+	priorityNormal eventPriority = iota
+	priorityLow
+)
+
+type alertType int
+
+const (
+	alertTypeSuccess alertType = iota
+	alertTypeInfo
+	alertTypeWarning
+	alertTypeError
+)
+
+type dogstatsdEvent struct {
+	title          []byte
+	text           []byte
+	timestamp      int64
+	hostname       []byte
+	aggregationKey []byte
+	priority       eventPriority
+	sourceType     []byte
+	alertType      alertType
+	tags           [][]byte
+}
+
+type eventHeader struct {
+	titleLength int
+	textLength  int
+}
+
+var (
+	eventTimestampPrefix      = []byte("d:")
+	eventHostnamePrefix       = []byte("h:")
+	eventAggregationKeyPrefix = []byte("k:")
+	eventPriorityPrefix       = []byte("p:")
+	eventSourceTypePrefix     = []byte("s:")
+	eventAlertTypePrefix      = []byte("t:")
+	eventTagsPrefix           = []byte("#")
+
+	eventPriorityLow    = []byte("low")
+	eventPriorityNormal = []byte("normal")
+
+	eventAlertTypeError   = []byte("error")
+	eventAlertTypeWarning = []byte("warning")
+	eventAlertTypeInfo    = []byte("info")
+	eventAlertTypeSuccess = []byte("success")
+)
+
+// splitHeaderEvent splits the event and the
+func splitHeaderEvent(message []byte) ([]byte, []byte, error) {
+	sepIndex := bytes.Index(message, colonSeparator)
+	if sepIndex == -1 {
+		return nil, nil, fmt.Errorf("invalid event: %q", message)
+	}
+	return message[:sepIndex], message[sepIndex+1:], nil
+}
+
+func parseHeader(rawHeader []byte) (eventHeader, error) {
+	if len(rawHeader) < 7 {
+		return eventHeader{}, fmt.Errorf("invalid event header: %q", rawHeader)
+	}
+	rawLengths := rawHeader[3 : len(rawHeader)-1]
+	sepIndex := bytes.Index(rawLengths, commaSeparator)
+	if sepIndex == -1 {
+		return eventHeader{}, fmt.Errorf("invalid event header: %q", rawHeader)
+	}
+	rawTitleLength := rawLengths[:sepIndex]
+	rawTextLength := rawLengths[sepIndex+1:]
+	titleLength, err := strconv.ParseInt(string(rawTitleLength), 10, 32)
+	if err != nil {
+		return eventHeader{}, fmt.Errorf("invalid event header: %q", rawHeader)
+	}
+	textLength, err := strconv.ParseInt(string(rawTextLength), 10, 32)
+	if err != nil {
+		return eventHeader{}, fmt.Errorf("invalid event header: %q", rawHeader)
+	}
+	return eventHeader{
+		titleLength: int(titleLength),
+		textLength:  int(textLength),
+	}, nil
+}
+
+func cleanEventText(text []byte) []byte {
+	return bytes.Replace(text, []byte("\\n"), []byte("\n"), -1)
```

Comment:
This may be somewhere we could add further improvement if needed: `bytes.Replace` returns a copy of the src string, because we're replacing with less chars, we could consider replace in-place to avoid an allocation.

Reply:
IMO we should leave that for later. Events are usually not a hot path